mev-lib/src/types/mod.rs

```diff
@@ -0,0 +1,197 @@
+pub mod bellatrix;
+pub mod capella;
+
+use crate::signing::{verify_signed_builder_message, verify_signed_consensus_message};
+pub use ethereum_consensus::builder::SignedValidatorRegistration;
+use ethereum_consensus::{
+    primitives::{BlsPublicKey, Hash32, Slot, ValidatorIndex},
+    state_transition::{Context, Error},
+};
+use ssz_rs::prelude::*;
+
+#[derive(Debug, Default, Clone, PartialEq, Eq, Hash)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+pub struct BidRequest {
+    #[serde(with = "crate::serde::as_string")]
+    pub slot: Slot,
+    pub parent_hash: Hash32,
+    pub public_key: BlsPublicKey,
+}
+
+#[derive(Debug)]
+pub enum BuilderBid {
+    Bellatrix(bellatrix::BuilderBid),
+    Capella(capella::BuilderBid),
+}
+
+#[derive(Debug, Clone)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+#[cfg_attr(feature = "serde", serde(tag = "version", content = "data"))]
+#[cfg_attr(feature = "serde", serde(rename_all = "lowercase"))]
+pub enum SignedBuilderBid {
+    Bellatrix(bellatrix::SignedBuilderBid),
+    Capella(capella::SignedBuilderBid),
+}
+
+impl SignedBuilderBid {
+    pub fn value(&self) -> &U256 {
+        match self {
+            Self::Bellatrix(bid) => &bid.message.value,
+            Self::Capella(bid) => &bid.message.value,
+        }
+    }
+
+    pub fn block_hash(&self) -> &Hash32 {
```

Comment:
not in love w/ this polymorphism strategy, but assuming nothing better it will make sense to consolidate this into a `derive` macro